server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -680,59 +680,66 @@ private List<String> compilePodSourceFiles(PodSpec pod, Map<String, Object> mani
         List<String> objs = new ArrayList<>();
         List<String> commands = new ArrayList<>();
 
-        // per supported language build lists of flags and defines
-        List<String> defines = new ArrayList<>(pod.defines);
-        List<String> flagsCpp = new ArrayList<>(pod.flags);
-        List<String> flagsObjC = new ArrayList<>(pod.flags);
-        List<String> flagsC = new ArrayList<>(pod.flags);
-        if (platform.contains("ios")) {
-            flagsCpp.addAll(pod.ios_flags);
-            flagsObjC.addAll(pod.ios_flags);
-            flagsC.addAll(pod.ios_flags);
-            defines.addAll(pod.ios_defines);
-        }
-        else if (platform.contains("osx")) {
-            flagsCpp.addAll(pod.osx_flags);
-            flagsObjC.addAll(pod.osx_flags);
-            flagsC.addAll(pod.osx_flags);
-            defines.addAll(pod.osx_defines);
-        }
-        removeFlag("-std=c++", flagsObjC);
-        removeFlag("-std=c++", flagsC);
+        // remove hardcoded flags from context
+        Map<String, Object> trimmedContext = ExtenderUtil.mergeContexts(manifestContext, new HashMap<>());
+        removeFlag("-fno-exceptions", (List<String>)trimmedContext.get("flags"));
+        removeFlag("-stdlib=libc++", (List<String>)trimmedContext.get("flags"));
 
         // create contexts per supported language
+        // add pod flags and defines
+        Map<String, Object> podContextC = new HashMap<>();
         Map<String, Object> podContextCpp = new HashMap<>();
         Map<String, Object> podContextObjC = new HashMap<>();
-        Map<String, Object> podContextC = new HashMap<>();
-        podContextCpp.put("flags", flagsCpp);
-        podContextCpp.put("defines", defines);
-        podContextObjC.put("flags", flagsObjC);
-        podContextObjC.put("defines", defines);
-        podContextC.put("flags", flagsC);
-        podContextC.put("defines", defines);
-        Map mergedContextWithPodsForCpp = ExtenderUtil.mergeContexts(manifestContext, podContextCpp);
-        Map mergedContextWithPodsForObjC = ExtenderUtil.mergeContexts(manifestContext, podContextObjC);
-        Map mergedContextWithPodsForC = ExtenderUtil.mergeContexts(manifestContext, podContextC);
+        Map<String, Object> podContextObjCpp = new HashMap<>();
+
+        if (platform.contains("ios")) {
```

Comment:
What are examples of these flags?
Is it a complete set?
Can we use them as-is?
Do we really need our compile flags?
Perhaps, if it's just paths we need, those paths should be better specified in out `build_input.yml`?

Reply:
I think you are right. The flags we get from the podspec should be a complete set so there's not really any need to merge them.